src/main/java/mate/academy/rickandmorty/service/CharacterServiceImpl.java

```diff
@@ -0,0 +1,38 @@
+package mate.academy.rickandmorty.service;
+
+import java.util.List;
+import java.util.Random;
+import lombok.RequiredArgsConstructor;
+import mate.academy.rickandmorty.dto.external.CharacterResponseDto;
+import mate.academy.rickandmorty.dto.internal.CharacterRequestDto;
+import mate.academy.rickandmorty.mapper.CharacterMapper;
+import mate.academy.rickandmorty.repository.CharacterRepository;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class CharacterServiceImpl implements CharacterService {
+    private final CharacterRepository characterRepository;
+    private final CharacterMapper mapper;
+
+    @Override
+    public CharacterRequestDto getRandomCharacter() {
+        Random random = new Random();
+        int id = random.nextInt((int) characterRepository.count());
+        return mapper.toDto(characterRepository.getReferenceById((long) id));
+    }
+
+    @Override
+    public List<CharacterRequestDto> findByName(String name) {
+        return characterRepository.findAllByNameContainsIgnoreCase(name).stream()
+                .map(mapper::toDto)
+                .toList();
+    }
+
+    @Override
+    public void saveAll(List<CharacterResponseDto> characterResponseDtos) {
+        for (CharacterResponseDto character : characterResponseDtos) {
+            characterRepository.save(mapper.toModel(character));
+        }

```

Comment:
let's optimize this by using `saveAll`
```suggestion
    List<Character> characters = characterResponseDtos.stream()
                                                      .map(mapper::toModel)
                                                      .collect(Collectors.toList());
    characterRepository.saveAll(characters);
```

Reply:
Tried to do this, but something went wrong, i think i didn't convert it to model before saving. Thanks for pointing that out fot me

Reply:
Oh, and one more thing. Will it be wrong, if i won't declare a variable and do all the operations inside the saveAll() operation?

Reply:
would be compiled for sure but
1) Not really the best structure to read
`
    characterRepository.saveAll(characterResponseDtos.stream()
                                                      .map(mapper::toModel)
                                                      .toList();
`
2) Harder to debug. You actually cannot place any valuable breakpoint so you will have to work with evaluate expression only